projects/plugins/jetpack/extensions/blocks/blogroll/utils.js

```diff
@@ -8,14 +8,15 @@ export function createBlockFromRecommendation( attrs ) {
 }
 
 export function createBlockFromSubscription( subscription ) {
-	const { blog_id, site_icon, URL, name, description } = subscription;
+	const { blog_id, id, site_icon, URL, name, description, is_non_wpcom_site } = subscription;
 
 	return createBlockFromRecommendation( {
-		id: blog_id,
```

Comment:
I'm seeing a 'Saving failed` error when I manually search and add a WPCOM site.
<img width="853" alt="CleanShot 2023-11-16 at 15 06 32@2x" src="https://github.com/Automattic/jetpack/assets/10482592/d5bcd833-fbd9-4a6b-a1a5-01f110738505">

I no longer see the error if I change this line from `id: blog_id || id` to `id: blog_id`. Do we need the additional `id` value?

Reply:
Thank you for reviewing, the issue was because sites coming from blogroll suggestions had their ID as a string and sites coming from search itself had them as integers. Apparently, it was not failing in the BE but showing this message in the FE. I pushed a change to threat all IDs as strings that fixed this issue.